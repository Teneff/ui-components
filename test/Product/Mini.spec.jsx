import React from 'react';
import { shallow } from 'enzyme';
import Mini from '../../components';

describe('<Mini />', () => {
  const props = {
    image:
      'https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd',
    name: 'London Chic Scuba Dress',
    link:
      '/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1',
    price: 20,
    rating: {
      value: 0.5,
      total: 10,
    },
  };

  const component = shallow(<Mini {...props} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
