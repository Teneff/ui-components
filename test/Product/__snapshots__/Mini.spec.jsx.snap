// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mini /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mini
    image="https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd"
    link="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
    name="London Chic Scuba Dress"
    price={20}
    rating={
      Object {
        "total": 10,
        "value": 0.5,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
          title="London Chic Scuba Dress"
        >
          <Image
            alt=""
            src="https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd"
          />
        </Link>,
        <Price
          value={20}
        />,
        <Link
          href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
          title="London Chic Scuba Dress"
        >
          <Name
            value="London Chic Scuba Dress"
          />
        </Link>,
        <Rating
          count={5}
          total={10}
          value={0.5}
        />,
        <div>
          <Link
            href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
            title="London Chic Scuba Dress"
          >
            <Button
              text="More Info"
            />
          </Link>
        </div>,
      ],
      "style": Object {
        "float": "left",
        "margin": "10px",
        "maxWidth": "300px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Image
            alt=""
            src="https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd"
          />,
          "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
          "title": "London Chic Scuba Dress",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "",
            "src": "https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "value": 20,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Name
            value="London Chic Scuba Dress"
          />,
          "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
          "title": "London Chic Scuba Dress",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "value": "London Chic Scuba Dress",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "count": 5,
          "total": 10,
          "value": 0.5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
            title="London Chic Scuba Dress"
          >
            <Button
              text="More Info"
            />
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              text="More Info"
            />,
            "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
            "title": "London Chic Scuba Dress",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "More Info",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
            title="London Chic Scuba Dress"
          >
            <Image
              alt=""
              src="https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd"
            />
          </Link>,
          <Price
            value={20}
          />,
          <Link
            href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
            title="London Chic Scuba Dress"
          >
            <Name
              value="London Chic Scuba Dress"
            />
          </Link>,
          <Rating
            count={5}
            total={10}
            value={0.5}
          />,
          <div>
            <Link
              href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
              title="London Chic Scuba Dress"
            >
              <Button
                text="More Info"
              />
            </Link>
          </div>,
        ],
        "style": Object {
          "float": "left",
          "margin": "10px",
          "maxWidth": "300px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Image
              alt=""
              src="https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd"
            />,
            "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
            "title": "London Chic Scuba Dress",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "",
              "src": "https://asda.scene7.com/is/image/Asda/5054532217318?hei=289&wid=218&qlt=85&fmt=pjpg&resmode=sharp&op_usm=1.1,0.5,0,0&defaultimage=default_details_George_rd",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "value": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Name
              value="London Chic Scuba Dress"
            />,
            "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
            "title": "London Chic Scuba Dress",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "value": "London Chic Scuba Dress",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 5,
            "total": 10,
            "value": 0.5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              href="/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1"
              title="London Chic Scuba Dress"
            >
              <Button
                text="More Info"
              />
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                text="More Info"
              />,
              "href": "/george/womens/dresses/london-chic-scuba-dress/GEM362216,default,pd.html?cgid=D1M1G20C1",
              "title": "London Chic Scuba Dress",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "More Info",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
